src/storage/store/engine.rs

```diff
@@ -349,6 +349,52 @@ impl ShardEngine {
         result
     }
 
+    fn txn_summary(txn: &Transaction) -> String {
+        let mut summary = String::new();
```

Comment:
I think the summary should include the fid once (each txn is per fid), and the identifier for each message type (message hash, txn hash and log index for onchain event and fname), every thing else we can determine based on the identifier.

Reply:
Fixed. I included some of the redundant info because I think it'll be useful for eyeballing the logs and seeing if there are patterns.